### PR TITLE
Native onchain development - Updated hello world lesson

### PR DESCRIPTION
**Pull request by 0xCipherCoder**
### Problem

Fixed grammatical mistakes, and updated code and styling as per guidelines.

### Summary of Changes

Unboxed frontend PR - https://github.com/Unboxed-Software/solana-hello-world-client/pull/5 

Fixes #

### CODE REVIEW

File: content/courses/native-onchain-development/hello-world-program.md [hunk 1]
```diff
@@ -13,156 +13,152 @@ description:
...
 </Callout>
 
-This lesson will give you a basic introduction to writing and deploying a Solana
-program using the Rust programming language, without any framework. This gives
-you greater control, but also requires you to perform much of the basic work of
-creating an onchain program yourself.
+This lesson will introduce you to writing and deploying a Solana program using
+the Rust programming language without any framework. This approach offers
+greater control but requires you to handle much of the foundational work of
+creating an on-chain program yourself.
```
Comment (mikemaccana) [inline content/courses/native-onchain-development/hello-world-program.md:32]: 'onchain' is one word. See https://github.com/solana-foundation/developer-content/blob/main/CONTRIBUTING.md#prose and fix everywhere.

File: content/courses/native-onchain-development/hello-world-program.md [hunk 2]
```diff
@@ -13,156 +13,152 @@ description:
 
 ## Summary
 
-- Native Solana programs have a single **entry point** to process instructions
-- A program processes an instruction using the **program_id**, list of
-  **accounts**, and **instruction_data** included with the instruction
+- Native Solana programs have a single **entry point** to process instructions.
+- A program processes an instruction using the **program_id**, a list of
+  **accounts**, and **instruction_data** included with the instruction.
```
Comment (mikemaccana) [inline content/courses/native-onchain-development/hello-world-program.md:18]: Love the attention to detail here. 👍

File: content/courses/native-onchain-development/hello-world-program.md [hunk 3]
```diff
@@ -13,156 +13,152 @@ description:
...
-This includes:
+system.” This includes:
 
-- **Modules** - A module separates code into logical units to provide isolated
-  namespaces for organization, scope, and privacy of paths
-- **Crates** - A crate is either a library or an executable program. The source
-  code for a crate is usually subdivided into multiple modules.
-- **Packages** - A package contains a collection of crates as well as a manifest
-  file for specifying metadata and dependencies between packages
+- **Modules**: Separate code into logical units to provide isolated namespaces
```
Comment (mikemaccana) [inline content/courses/native-onchain-development/hello-world-program.md:48]:
```suggestion
- **Modules**: Separates code into logical units to provide isolated namespaces
```

Comment (mikemaccana) [reply content/courses/native-onchain-development/hello-world-program.md:48]: Small accident, but the rest of your writing is excellent.

File: content/courses/native-onchain-development/hello-world-program.md [hunk 4]
```diff
@@ -13,156 +13,152 @@ description:
...
 
-#### Paths and scope
+#### Paths and Scope
 
-Crates in Rust contain modules that define functionality which can be shared
-with multiple projects. If we want to access an item within a module, then we
-need to know its "path" (like when we're navigating a filesystem).
+Crates in Rust contain modules that define functionality, which can be shared
+across multiple projects. If we want to access an item within a module, we need
+to know its "path," similar to navigating a filesystem.
```
Comment (mikemaccana) [inline content/courses/native-onchain-development/hello-world-program.md:61]:
```suggestion
Crates contain modules that can be shared
across multiple projects. If we want to access an item within a module, we need
to know its "path," similar to navigating a filesystem.
```

Comment (mikemaccana) [reply content/courses/native-onchain-development/hello-world-program.md:61]: Yours was better that the original but we can go further.

File: content/courses/native-onchain-development/hello-world-program.md [hunk 5]
```diff
@@ -13,156 +13,152 @@ description:
...
 ```
 
-We can then have our functions return values by declaring the return type using
-an arrow `->` after the function.
+We can also have our functions return values by declaring the return type using
+an arrow -> after the function.
 
 In the example below, the `process_instruction` function will now return a value
-of type `ProgramResult`. We will go over this in the next section.
+of type `ProgramResult`. We’ll go over this in the next section.
```
Comment (mikemaccana) [inline content/courses/native-onchain-development/hello-world-program.md:140]:
```suggestion
of type `ProgramResult`. We'll go over this in the next section.
```

Comment (mikemaccana) [reply content/courses/native-onchain-development/hello-world-program.md:140]: I'm not totally sure there's any benefit to smart quotes. The rest of the code uses regular quotes so I think we should stick to that, or change it everywhere. Feel free to change my mind!

File: content/courses/native-onchain-development/hello-world-program.md [hunk 6]
```diff
@@ -205,72 +200,72 @@ use solana_program::{
...
-  addresses as a public key
-- `msg` - a macro that allows us to print messages to the program log
+- `AccountInfo` - A struct within the `account_info` module that allows us to
+  access account information.
+- `entrypoint` -A macro that declares the entry point of the program.
+- `ProgramResult` - A type within the `entrypoint` module that returns either a
+  `Result` or `ProgramError`.
+- `Pubkey` - A struct within the `pubkey` module that allows us to access
+  addresses as public keys.
+- `msg` - A macro that allows us to print messages to the program log.
```
Comment (mikemaccana) [inline content/courses/native-onchain-development/hello-world-program.md:210]: `msg` and `Pubkey` and `ProgramResult` are good. But `AccountInfo` and `entrypoint` are recursive. What is an entrypoint? What is account information?

> `entrypoint` - A macro that declares the function that receives incoming instructions and passes them on the right instruction handler.
 > `AccountInfo` - A struct that allows us to access account information like...

For example. You're a great writer I'm sure you can do better!

File: content/courses/native-onchain-development/hello-world-program.md [hunk 7]
```diff
@@ -317,78 +312,76 @@ use solana_program::{
...
+using Solana Playground.
 
 ```typescript
 let programId = new web3.PublicKey("<YOUR_PROGRAM_ID>");
 ```
 
-You can locate the program ID on Solana Playground referencing the image below.
+You can find the program ID on Solana Playground as shown below.
 
 ![Gif Solana Playground Program ID](/public/assets/courses/unboxed/hello-world-program-id.gif)
```
Comment (mikemaccana) [inline content/courses/native-onchain-development/hello-world-program.md:349]: You might want to make a new version of this image (or an mp4) if Solana Playground looks out of date. You decide.

File: content/courses/native-onchain-development/hello-world-program.md [hunk 8]
```diff
@@ -317,78 +312,76 @@ use solana_program::{
...
 
-Now, go ahead and run `npm start`. This command will:
+Afterwards, execute `npm start`. This command will:
 
-1. Generate a new keypair and create a `.env` file if one does not already exist
-2. Airdrop devnet SOL
-3. Invoke the “Hello, world!” program
-4. Output the transaction URL to view on Solana Explorer
+1. Generate a new keypair and create a `.env` file if it doesn't already exist.
+2. Airdrop some SOL on the Devnet.
```
Comment (mikemaccana) [inline content/courses/native-onchain-development/hello-world-program.md:356]:
```suggestion
2. Airdrop some SOL onto this account on devnet.
```

not 'the devnet'.

File: content/courses/native-onchain-development/hello-world-program.md [hunk 9]
```diff
@@ -317,78 +312,76 @@ use solana_program::{
...
+3. Invoke the “Hello, world!” program.
+4. Output a transaction URL that you can view on Solana Explorer.
 
-Copy the transaction URL printed in the console into your browser. Scroll down
-to see “Hello, world!” under Program Instruction Logs.
+Copy the transaction URL from the console into your browser. Scroll down to the
+Program Instruction Logs section to see “Hello, world!” displayed.
 
 ![Screenshot Solana Explorer Program Log](/public/assets/courses/unboxed/hello-world-program-log.png)
```
Comment (mikemaccana) [inline content/courses/native-onchain-development/hello-world-program.md:363]: Likewise you may want to update this screenshot if you think it's necessary.

File: content/courses/native-onchain-development/hello-world-program.md [hunk 10]
```diff
@@ -317,78 +312,76 @@ use solana_program::{
...
+1. Write a new program that uses the `msg!` macro to print your own custom
+   message to the program log.
+2. Build and deploy your program just like we did in the lab.
+3. Invoke your newly deployed program and use Solana Explorer to confirm that
+   your message was printed in the program log.
 
-As always, get creative with these challenges and take them beyond the basic
-instructions if you want - and have fun!
+As always, feel free to get creative with these challenges and go beyond the
+basic instructions if you want—most importantly, have fun!
```
Comment (mikemaccana) [inline content/courses/native-onchain-development/hello-world-program.md:381]:
```suggestion
basic instructions if you want — most importantly, have fun!
```

**Review by mikemaccana — CHANGES_REQUESTED** [review]
Excellent work with a huge amount of attention to detail. Fix the items above and we can get this in. You're the only person competing for this lesson right now so you have a good change to win it.

File: content/courses/native-onchain-development/hello-world-program.md (hunk 5 above)
Comment (0xCipherCoder) [reply content/courses/native-onchain-development/hello-world-program.md]: This was from old content only, I didn't notice. I have changed it to regular quotes across the lesson.

File: content/courses/native-onchain-development/hello-world-program.md (hunk 2 above)
Comment (0xCipherCoder) [reply content/courses/native-onchain-development/hello-world-program.md:18]: Thank you! @mikemaccana

File: content/courses/native-onchain-development/hello-world-program.md (hunk 6 above)
Comment (0xCipherCoder) [reply content/courses/native-onchain-development/hello-world-program.md]: Added more details as per suggestion to explain better.

File: content/courses/native-onchain-development/hello-world-program.md (hunk 7 above)
Comment (0xCipherCoder) [reply content/courses/native-onchain-development/hello-world-program.md:352]: I have not updated the new version as there is not much difference in UI and UX.

File: content/courses/native-onchain-development/hello-world-program.md (hunk 9 above)
Comment (0xCipherCoder) [reply content/courses/native-onchain-development/hello-world-program.md:366]: The same has not been updated to the new version as there is not much difference in UI and UX except for the addition of 2 new icons for tutorials and programs.

**Review by mikemaccana — APPROVED** [review]
Another win @0xCipherCoder ! 🏆 

Can you please double check I have moved any outstanding unboxed repos to solana-developers and merged all your PRs to them, and then confirm here?

Also feel free to say hi on telegram and twitter, I'm @mikemaccana

**Comment by 0xCipherCoder** [comment]
Thanks! @mikemaccana 

Sharing all details here - 

Program Security Signer auth - PRs are merged and part of solana-developers -[Starter](https://github.com/solana-developers/signer-auth/tree/starter) | [Solution](https://github.com/solana-developers/signer-auth/tree/solution) 

Program Security PDA Sharing -  PRs are merged and part of solana-developers - [Starter](https://github.com/solana-developers/pda-sharing/tree/starter) | [Solution](https://github.com/solana-developers/pda-sharing/tree/solution)

Onchain Anchor CPI - [Movie Review Program](https://github.com/Unboxed-Software/anchor-movie-review-program) - Not moved to solana-developers yet - I have 2 PRs raised for them - [Anchor PDA solution](https://github.com/Unboxed-Software/anchor-movie-review-program/pull/16)  | [Anchor CPI Solution - add-tokens](https://github.com/Unboxed-Software/anchor-movie-review-program/pull/17)

Onchain Intro to Anchor - Already part of solana-developers but PRs not merged yet - [Anchor counter increment solution](https://github.com/solana-developers/anchor-counter-program/pull/1) | [Anchor counter decrement solution](https://github.com/solana-developers/anchor-counter-program/pull/2)

Program Security Closing Account - Already part of solana-developers but PRs not merged yet - [Starter](https://github.com/solana-developers/closing-accounts/pull/1) | [Solution](https://github.com/solana-developers/closing-accounts/pull/2)

Program Security Bump Seed Canonicalization - [Bump Seed Canonicalization Repo](https://github.com/Unboxed-Software/solana-bump-seed-canonicalization)  - Not moved to solana-developers yet - I have 2 PRs raised for this  - [Starter](https://github.com/Unboxed-Software/solana-bump-seed-canonicalization/pull/2) | [Solution](https://github.com/Unboxed-Software/solana-bump-seed-canonicalization/pull/3)

Program Security Arbitrary CPI - Already part of solana-developers but PRs not merged yet - [Starter](https://github.com/solana-developers/arbitrary-cpi/pull/1) | [Solution](https://github.com/solana-developers/arbitrary-cpi/pull/2)

Program Security Account Data Matching - Already part of solana-developers but PRs not merged yet - [Starter](https://github.com/solana-developers/account-data-matching/pull/2) | [Solution](https://github.com/solana-developers/account-data-matching/pull/3)

Program Security Owner Check - [Owner Check Repo](https://github.com/Unboxed-Software/solana-owner-checks) -  Not moved to solana-developers yet - I have 2 PRs raised for this - [Starter](https://github.com/Unboxed-Software/solana-owner-checks/pull/5) | [Solution](https://github.com/Unboxed-Software/solana-owner-checks/pull/6)

Program Security Type Cosplay -[ Type cosplay Repo](https://github.com/Unboxed-Software/solana-type-cosplay) - Not moved to solana-developers yet but PRs are merged

Hello world client - [Hello world repo](https://github.com/solana-developers/hello-world-client) - Already merged and part of solana-developers

Program Security Duplicate Mutable Accounts - [Repository](https://github.com/solana-developers/duplicate-mutable-accounts) - Already merged and part of solana-developers

Program Security Reinitialization Attack - Already part of solana-developers but I need to rebase original PRs due to conflict but PRs - [Starter](https://github.com/Unboxed-Software/solana-reinitialization-attacks/pull/4) | [Solution](https://github.com/Unboxed-Software/solana-reinitialization-attacks/pull/5)

Let me know if anything missed I am also checking.

**Pull request merged**